### ISSUE TITLE
new biopython version incompatible with HiCExplorer

### ISSUE THREAD

**Issue by bgruening** [deeptools/HiCExplorer#691]
**Welcome to the HiCExplorer GitHub repository! Before opening the issue please check
 that the following requirements are met :**

- [x] Search whether this issue (or a similar issue) has been solved before using the search tab above. Link the previous issue if appropriate below.

- [x] Paste your HiCExplorer version (`hicInfo --version`) and your python version (`python --version`) below. 
- [x] Have you checked our documentation on [hicexplorer.readthedocs.io](https://hicexplorer.readthedocs.io)?
- [x] Do you use [conda](https://docs.conda.io/projects/conda/en/latest/user-guide/install/linux.html) to install HiCExplorer? 
- [x] Do you use the latest HiCExplorer release? If not, please install it via a conda environment: 
       `conda create --name hicexplorer hicexplorer=3.6 python=3.8 -c bioconda -c conda-forge`
       and activate the environment: `conda activate hicexplorer`.  Retry your command. You can exit a conda environment via `conda deactivate`. To learn more about conda and environments, please consider the following [documentation](https://docs.conda.io/projects/conda/en/latest/user-guide/getting-started.html#).

The following is not supported anymore with the latest Biopthon release.

```python
from Bio.Alphabet import generic_dna
````
And will results in a Traceback like:

```
Traceback (most recent call last):
  File "/home/bag/miniconda3/envs/mulled-v1-09ea340851a0a0d219d489590b2fe9df151628584a390fbd5b5e08cdd905038f/bin/hicBuildMatrix", line 4, in <module>
    from hicexplorer.hicBuildMatrix import main
  File "/home/bag/miniconda3/envs/mulled-v1-09ea340851a0a0d219d489590b2fe9df151628584a390fbd5b5e08cdd905038f/lib/python3.8/site-packages/hicexplorer/hicBuildMatrix.py", line 24, in <module>
    from Bio.Alphabet import generic_dna
  File "/home/bag/.local/lib/python3.8/site-packages/Bio/Alphabet/__init__.py", line 20, in <module>
    raise ImportError(
ImportError: Bio.Alphabet has been removed from Biopython. In many cases, the alphabet can simply be ignored and removed from scripts. In a few cases, you may need to specify the ``molecule_type`` as an annotation on a SeqRecord for your script to work correctly. Please see https://biopython.org/wiki/Alphabet for more information.
```

```bash
(mulled-v1-09ea340851a0a0d219d489590b2fe9df151628584a390fbd5b5e08cdd905038f) bag@bag:~/miniconda3/envs/__hicexplorer@3.6/lib$ hicexplorer --version
hicexplorer 3.6
(mulled-v1-09ea340851a0a0d219d489590b2fe9df151628584a390fbd5b5e08cdd905038f) bag@bag:~/miniconda3/envs/__hicexplorer@3.6/lib$ python --version
Python 3.8.5

```

**Comment by joachimwolff** [comment]
For this reason the biopython version is restricted to be smaller 1.77: https://github.com/deeptools/HiCExplorer/blob/master/requirements.txt#L8

https://github.com/bioconda/bioconda-recipes/blob/master/recipes/hicexplorer/meta.yaml#L28

However, even if we want to fix it, we need to wait for cooler to update to the new version: https://github.com/bioconda/bioconda-recipes/blob/master/recipes/cooler/meta.yaml#L24

**Comment by bgruening** [comment]
This is still true and refers to the code. I think it makes sense to leave it open until the code supports newer versions of biopython. Or did you fixed that in dev?

**Comment by NessieCode** [comment]
Hello Maintainers of HiCExplorer

using the following provided command for new setup/installs 
conda create --name hicexplorer hicexplorer=3.6 python=3.8 -c bioconda -c conda-forge

Returns package python conflicts for:
hicexplorer=3.6 -> python[version='>=3.6']
python=3.8
hicexplorer=3.6 -> biopython[version='<1.77'] -> python[version='2.7.*|3.5.*|3.6.*|3.4.*|>=3.6,<3.7.0a0|>=2.7,<2.8.0a0|>=3.8,<3.9.0a0|>=3.7,<3.8.0a0|>=3.5,<3.6.0a0|<3.7|>=3.9,<3.10.0a0|>=3|<3.0.0|>=2.7|2.7|>=3.5']


This github issue seems to be related but not explicitly mentions new setups/installs

can you please advise on a possible workaround for this maybe a version of hicexplorer that still works for new installs,

**Comment by joachimwolff** [comment]
Hm, that worked today for me... Maybe some dependency was updated and causes now a conflict. Can you try the same command but add ˋcooler=0.8.9ˋ? As far as I know they had version 0.9 in the pipeline and that is the package which had biopython fixed to 1.77 so far.

**Comment by NessieCode** [comment]
using "conda create --name hicexplorer hicexplorer=3.6 cooler=0.8.9 python=3.8 -c bioconda -c conda-forge" increases the number of conflicts, but now I believe this might be related to the conda version, I did a fresh install of anaconda3-2020.11 on a test vm and installed without issue, the cluster I'm having an issue on is running a older version 4.8.4

Package python conflicts for:
cooler=0.8.9 -> python
hicexplorer=3.6 -> biopython[version='<1.77'] -> python[version='2.7.*|3.5.*|3.6.*|3.4.*|>=3.6,<3.7.0a0|>=2.7,<2.8.0a0|>=3.8,<3.9.0a0|>=3.7,<3.8.0a0|>=3.5,<3.6.0a0|<3.7|>=3.9,<3.10.0a0|>=3|<3.0.0|>=2.7|2.7|>=3.5']
hicexplorer=3.6 -> python[version='>=3.6']
cooler=0.8.9 -> asciitree -> python[version='2.7.*|3.5.*|3.6.*|3.4.*|>=3.6,<3.7.0a0|>=2.7,<2.8.0a0|>=3.8,<3.9.0a0|>=3.7,<3.8.0a0|>=3.5,<3.6.0a0|>=3.9,<3.10.0a0|>=3.7|>=3.6|>=3.5']
python=3.8

Package setuptools conflicts for:
hicexplorer=3.6 -> hyperopt[version='>=0.2.4'] -> setuptools
python=3.8 -> pip -> setuptools

Package ca-certificates conflicts for:
cooler=0.8.9 -> python -> ca-certificates
python=3.8 -> openssl[version='>=1.1.1j,<1.1.2a'] -> ca-certificates

Package cooler conflicts for:
hicexplorer=3.6 -> cooler[version='>=0.8.10']
cooler=0.8.9
hicexplorer=3.6 -> hic2cool[version='>=0.8.3'] -> cooler[version='>=0.8.5']

**Comment by joachimwolff** [comment]
I think the issue needs be solved by updating your conda installation, maybe you can report this as a bug to conda itself. 
The only idea I have to get it installed is to try a older python version, maybe 3.6 or 3.7 work with the outdated conda version?

**Comment by joachimwolff** [comment]
Biopython version fixed by Lucille PR #718.
